pkg/metallb/bgppeer.go

```diff
@@ -261,6 +261,33 @@ func (builder *BGPPeerBuilder) Update(force bool) (*BGPPeerBuilder, error) {
 	return builder, err
 }
 
+// WithDynamicASN defines the dynamicASN as either internal (iBGP) or external (eBGP). Both remoteAS and dynamicASN
+// configure the remote ASN. They are mutually exclusive and only one can be used per remote peer.
+func (builder *BGPPeerBuilder) WithDynamicASN(dynamicASN mlbtypesv1beta2.DynamicASNMode) *BGPPeerBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Creating BGPPeer %s in namespace %s using a dynamicASN: %s",
+		builder.Definition.Name, builder.Definition.Namespace, dynamicASN)
+
+	if dynamicASN != "internal" && dynamicASN != "external" {
+		glog.V(100).Infof("The dynamicASN of the BGPPeer is incorrect")
+
+		builder.errorMsg = "BGPPeer 'dynamicASN' must be either internal or external"
```

Comment:
```suggestion
		builder.errorMsg = "bGPPeer 'dynamicASN' must be either internal or external"
```
error messages should start with a lowercase letter

Reply:
changed